pkg/azuredx/azureauth/auth.go

```diff
@@ -147,71 +141,10 @@ func (c *ServiceCredentialsImpl) queryDataOnBehalfOf(ctx context.Context, req *b
 	return "Bearer " + onBehalfOfToken, nil
 }
 
-// OnBehalfOf resolves a token which impersonates the subject of userToken.
-// UserToken has to be an ID token. See the following link for more detail.
-// https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow
-func (c *ServiceCredentialsImpl) onBehalfOf(ctx context.Context, userToken string) (onBehalfOfToken string, expire time.Time, err error) {
-	params := make(url.Values)
-	params.Set("client_id", c.ClientID)
-	params.Set("client_secret", c.Secret)
-	params.Set("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer")
-	params.Set("assertion", userToken)
-	params.Set("scope", strings.Join(c.scopes, " "))
-	params.Set("requested_token_use", "on_behalf_of")
-	reqBody := strings.NewReader(params.Encode())
-
-	// https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-v2-protocols#endpoints
-	tokenURL := fmt.Sprintf("%s%s/oauth2/v2.0/token", c.authority, url.PathEscape(c.TenantID))
-	req, err := http.NewRequestWithContext(ctx, "POST", tokenURL, reqBody)
-	if err != nil {
-		return "", time.Time{}, fmt.Errorf("on-behalf-of grant request <%q> instantiation: %w", tokenURL, err)
-	}
-	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	req.Header.Set("Accept", "application/json")
-
-	// HTTP Exchange
-	reqStart := time.Now()
-	resp, err := c.HTTPDo(req)
+func (c *ServiceCredentialsImpl) onBehalfOf(ctx context.Context, idToken string) (onBehalfOfToken string, expire time.Time, err error) {
+	result, err := c.aadClient.AcquireTokenOnBehalfOf(ctx, idToken, c.scopes)
 	if err != nil {
-		return "", time.Time{}, fmt.Errorf("on-behalf-of grant POST <%q>: %w", tokenURL, err)
-	}
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return "", time.Time{}, fmt.Errorf("on-behalf-of grant POST <%q> response: %w", tokenURL, err)
-	}
-	onBehalfOfLatencySeconds.WithLabelValues(strconv.Itoa(resp.StatusCode)).Observe(float64(time.Since(reqStart)) / float64(time.Second))
```

Comment:
I am not sure if this was intended but you are removing the metric here. I am not sure if this metric is useful though but if we want to remove it, we should also remove the variable `onBehalfOfLatencySeconds`. If we remove it we should also make a note in the changelog. What do you think?

Reply:
I removed it intentionally because a single metric doesn't bring good picture and I didn't understand how it could be useful.
We could add metrics on the  Grafana Azure SDK layer, so they could be transparent to datasource and cover different scenarios (different auth types, cache hits etc.).

Removed references to the metric and updated the CHANGELOG.

Added task to Grafana Azure SDK: https://github.com/grafana/grafana-azure-sdk-go/issues/7

Reply:
okay, sounds good